src/containers/BookList.js

```diff
@@ -1,40 +1,61 @@
 import React from 'react';
 import propTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { removeBook } from '../actions';
+import { removeBook, changeFilter } from '../actions';
 import Book from '../component/Book';
+import CategoryFilter from '../component/CategoryFilter';
 
-const BookList = ({ books: { books }, removeBook }) => {
+const BookList = ({
+  books: { books }, filter: { filter }, removeBook, changeCategoryFilter,
+}) => {
   const handleRemoveBook = book => {
     removeBook(book);
   };
 
   const callBook = book => (<Book book={book} handler={handleRemoveBook} />);
+  const changeFilter = key => {
+    changeCategoryFilter(key);
+  };
+
+  const filteredBooks = () => {
+    if (filter !== 'All') {
+      return books.filter(book => book.category === filter);
+    }
+
+    return books;
+  };
 
   return (
-    <table>
-      <thead>
-        <tr>
-          <td>ID</td>
-          <td>TITLE</td>
-          <td>CATEGORY</td>
-        </tr>
-      </thead>
-      <tbody>
-        {books.map(callBook)}
-      </tbody>
-    </table>
+    <div>
+      <CategoryFilter updateFilter={changeFilter} />
+      <table>
+        <thead>
+          <tr>
+            <td>ID</td>
+            <td>TITLE</td>
+            <td>CATEGORY</td>
+          </tr>
+        </thead>
+        <tbody>
+          {filteredBooks().map(callBook)}
+        </tbody>
+      </table>
+    </div>
   );
 };
 
 const mapStateToProps = state => ({
   books: state.books,
+  filter: state.filter,
```

Comment:
I wouldn't recommend using redux this way, instead, why not return the filtered books in the `mapStateToProps` function?